### PR TITLE
pgmq: init at 1.4.4

### PR DESCRIPTION
**Pull request by takeda**
PGMQ is a message queue extension that allows to implement similar queuing that's available on AWS SQS or RSMQ.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/sql/postgresql/ext/pgmq.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  postgresql,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pgmq";
+  version = "1.4.4";
+
+  src = fetchFromGitHub {
+    owner = "tembo-io";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/pgmq.nix:14]:
```suggestion
    repo = "pgmq";
```

File: pkgs/servers/sql/postgresql/ext/pgmq.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    rev = "v${version}";
+    hash = "sha256-z+8/BqIlHwlMnuIzMz6eylmYbSmhtsNt7TJf/CxbdVw=";
+  };
+
+  sourceRoot = "${src.name}/${pname}-extension";
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/pgmq.nix:19]:
```suggestion
  sourceRoot = "${src.name}/pgmq-extension";
```

File: pkgs/servers/sql/postgresql/ext/pgmq.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  buildInputs = [ postgresql ];
+
+  installPhase = ''
+    install -D -t $out/share/postgresql/extension sql/*.sql
+    install -D -t $out/share/postgresql/extension *.control
+  '';
+
+  meta = with lib; {
+    description = "A lightweight message queue. Like AWS SQS and RSMQ but on Postgres.";
```
Comment (SuperSandro2000) [inline pkgs/servers/sql/postgresql/ext/pgmq.nix:31]:
```suggestion
    description = "Lightweight message queuel ike AWS SQS and RSMQ but on Postgres";
```

**Comment by takeda** [comment]
@SuperSandro2000 applied proposed fixes.

File: pkgs/servers/sql/postgresql/ext/pgmq.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    install -D -t $out/share/postgresql/extension *.control
+  '';
+
+  meta = with lib; {
+    description = "Lightweight message queue like AWS SQS and RSMQ but on Postgres";
+    homepage = "https://tembo.io/pgmq";
+    maintainers = with maintainers; [ takeda ];
+    platforms = postgresql.meta.platforms;
+    license = licenses.postgresql;
+  };
```
Comment (zimward) [inline pkgs/servers/sql/postgresql/ext/pgmq.nix:36]:
```suggestion
  meta = {
    description = "Lightweight message queue like AWS SQS and RSMQ but on Postgres";
    homepage = "https://tembo.io/pgmq";
    maintainers = with lib.maintainers; [ takeda ];
    platforms = postgresql.meta.platforms;
    license = lib.licenses.postgresql;
  };
```
See: #208242

File: pkgs/servers/sql/postgresql/ext/pgmq.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  sourceRoot = "${src.name}/pgmq-extension";
+
+  dontConfigure = true;
+
+  buildInputs = [ postgresql ];
+
+  installPhase = ''
+    install -D -t $out/share/postgresql/extension sql/*.sql
+    install -D -t $out/share/postgresql/extension *.control
+  '';
```
Comment (h7x4) [inline pkgs/servers/sql/postgresql/ext/pgmq.nix:28]:
```suggestion
  installPhase = ''
    runHook preInstall
    
    install -D -t $out/share/postgresql/extension sql/*.sql
    install -D -t $out/share/postgresql/extension *.control
    
    runHook postInstall
  '';
```

File: pkgs/servers/sql/postgresql/ext/pgmq.nix [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+  meta = with lib; {
+    description = "Lightweight message queue like AWS SQS and RSMQ but on Postgres";
+    homepage = "https://tembo.io/pgmq";
```
Comment (h7x4) [inline pkgs/servers/sql/postgresql/ext/pgmq.nix:32]:
```suggestion
    homepage = "https://tembo.io/pgmq";
    changelog = "https://github.com/tembo-io/pgmq/releases/tag/v${version}";
```

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -20711,6 +20711,12 @@
     githubId = 18656090;
     name = "Yuki Takagi";
   };
+  takeda = {
+    name = "Derek Kuliński";
+    email = "d@kulinski.us";
+    github = "takeda";
+    githubId = 411978;
+  };
```
Comment (zimward) [inline maintainers/maintainer-list.nix:20719]: This should be a seperate commit: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

**Review by h7x4 — COMMENTED** [review]
Could you squash and fix up your commits so that there are two commits like this?

```
maintainers: add takeda
pgmq: init at 1.4.4
```

**Review by zimward — COMMENTED** [review]
Please also squash fix commits.

**Comment by takeda** [comment]
@h7x4 @zimward I applied your recommendations

**Review by zimward — COMMENTED** [review]
I think it should be good to go, but i know too little about `postgresql` extensions so somebody more knowledgeable then me should probably look over it one more time.

**Pull request merged**